common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1527,65 +1527,55 @@ bool ap_wlan_hal_dwpal::read_acs_report()
-    char *p2str     = nullptr;
-    char *lineMsg   = strtok_s(reply, &dmaxLen, "\n", &p2str);
-    int i           = 0;
-    while (lineMsg != NULL) {
-        size_t lineLen = strnlen(lineMsg, HOSTAPD_TO_DWPAL_MSG_LENGTH);
-
-        // TODO: fieldsToParse.field should be contained in one struct and struct size should be
-        //       passed to the dwpal_string_to_struct_parse - a security fix for multiline reply
-        if (dwpal_string_to_struct_parse(lineMsg, lineLen, fieldsToParse, sizeof(dfs_state)) ==
-            DWPAL_FAILURE) {
-            LOG(ERROR) << "DWPAL parse error ==> Abort";
-            return false;
+    int idx = 0;
+    for (auto band : radio_info.bands) {
+        auto supported_channels = band.supported_channels;
+        std::unordered_map<uint8_t, bwl::nl80211_client::channel_info>::iterator it;
+        for (it = supported_channels.begin(); it != supported_channels.end(); ++it) {
+            auto channel_info = it->second;
+            for (auto bw : channel_info.supported_bandwidths) {
+                auto channel_info_ptr     = &m_radio_info.supported_channels[idx++];
+                channel_info_ptr->channel = channel_info.number;
+                channel_info_ptr->tx_pow  = channel_info.tx_power;
+                channel_info_ptr->is_dfs  = channel_info.is_dfs;
+                switch (bw) {
+                case beerocks::eWiFiBandwidth::BANDWIDTH_20:
+                    channel_info_ptr->bandwidth = 20;
+                    break;
+                case beerocks::eWiFiBandwidth::BANDWIDTH_40:
+                    channel_info_ptr->bandwidth = 40;
+                    break;
+                case beerocks::eWiFiBandwidth::BANDWIDTH_80:
+                    channel_info_ptr->bandwidth = 80;
+                    break;
+                case beerocks::eWiFiBandwidth::BANDWIDTH_80_80:
+                    channel_info_ptr->bandwidth = 160;
+                    break;
+                case beerocks::eWiFiBandwidth::BANDWIDTH_160:
+                    channel_info_ptr->bandwidth = 160;
+                    break;
+                case beerocks::eWiFiBandwidth::BANDWIDTH_MAX:
+                case beerocks::eWiFiBandwidth::BANDWIDTH_UNKNOWN:
+                    channel_info_ptr->bandwidth = 0;
+                    break;
+                }
+            }
         }
-
-        // LOG(DEBUG) << "numOfValidArgs[0]= " << numOfValidArgs[0] << " chan= " << chan;
-        // LOG(DEBUG) << "numOfValidArgs[1]= " << numOfValidArgs[1] << " dfs_state= " << dfs_state;
-
-        m_radio_info.supported_channels[i].bandwidth   = 20;
-        m_radio_info.supported_channels[i].channel     = chan;
-        m_radio_info.supported_channels[i].bss_overlap = 0;
-        m_radio_info.supported_channels[i].is_dfs      = numOfValidArgs[1];
-
-        i++;
-
-        chan = -1;
-        memset(dfs_state, 0, sizeof(dfs_state));
-
-        lineMsg = strtok_s(NULL, &dmaxLen, "\n", &p2str);
     }
-
-    return true;
+    return ret;
```

Comment:
Don't return ret, return `true` or false. This is redundant.

Reply:
Changed